### PR TITLE
NFV update workload scripts changes

### PR DESCRIPTION
**Pull request by Jaganathancse**
This change is to add workload automation support for NFV OVS DPDK.

### CODE REVIEW

**Comment by lewisdenny** [comment]
recheck
`build-push-container-cifmw-client` failed due to unrelated issue to this patch.

**Review by rdiazcam — APPROVED** [review]
lgtm

**Comment by cjeanner** [comment]
/appove
/lgtm

NFV team tested it.

File: roles/update/templates/workload_launch.sh.j2 [hunk 1]
```diff
@@ -319,6 +324,9 @@ function workload_launch {
             --ram {{ workload_memory|default(512) }} \
             --disk {{ workload_disk|default(10) }} \
             --swap {{ workload_swap|default(1) }} \
+            --property hw:mem_page_size=large \
```
Comment (ciecierski) [inline roles/update/templates/workload_launch.sh.j2:327]: For non-nfv workload I don't think we should set this properties.
I suggest to create nfv flavor and non-nfv inside if-else condition

Comment (Jaganathancse) [reply roles/update/templates/workload_launch.sh.j2:327]: @ciecierski ack

Comment (Jaganathancse) [reply roles/update/templates/workload_launch.sh.j2]: done

File: roles/update/templates/workload_launch.sh.j2 [hunk 2]
```diff
@@ -188,6 +189,8 @@ function sanity_teardown {
     local elapsed_seconds=0
     {% if workload_sriov|default(false) | bool -%}
     openstack port delete "${SRIOV_PORT}"
+    {% elif workload_dpdk|default(false) | bool -%}
+    openstack port delete "${DPDK_PORT}"
     {% else -%}
```
Comment (sathlan) [inline roles/update/templates/workload_launch.sh.j2:194]: Oki, so I'm not an expert in `dpdk` - or `sriov` for that matter - but just to make sure.  Here (at L193) we don't remove the FIP from the INSTANCE.

But on L50 of that same file (`set_vm_ip` function called on L438), we do create a FIP.  So it means that just deleting the port remove the FIP.  Is that correct ?  Also, also around line 50, for `sriov `the command to get FIP was different, in the sense that creating the port seems to create the FIP.  Here for `dpdk` we don't have any special treatment. 

Overall, that looks strange to me: we only need to delete the port to get rid of the FIP, but we do have to run `openstack floating ip create` to create it.

So is that ok and it just shows my lack of knowdegde in dpdk or is there something wrong there ?

Thanks,

**Review by sathlan — COMMENTED** [review]
Hi,

the destruction of the FIP only need the destruction of the port, but the creation of the FIP needs the `openstack floating ip create`. Is that correct ?

Thanks,

**Comment by sathlan** [comment]
lgtm, @cjeanner I think there's no more comment from the update team on this one.

**Comment by Jaganathancse** [comment]
> Hi,
> 
> the destruction of the FIP only need the destruction of the port, but the creation of the FIP needs the `openstack floating ip create`. Is that correct ?
> 
> Thanks,

@sathlan  
correct,  FIP deletion part is also already handled in latest code. 
https://github.com/openstack-k8s-operators/ci-framework/blob/main/roles/update/templates/workload_launch.sh.j2#L198

so we can merge this PR.

**Comment by viroel** [comment]
/lgtm

code change is protected with `workload_dpdk` var, which defaults to `false`. looks safe to merde

**Pull request merged**